activerecord/lib/active_record/relation/batches.rb

```diff
@@ -5,6 +5,7 @@
 module ActiveRecord
   # = Active Record \Batches
   module Batches
+    class UnpersistedRecordError < ActiveRecordError; end
```

Comment:
@nvasilevski let's pass the unpersisted record to `UnpersistedRecordError`. if a user were to catch it, they would want to know which specific record caused the raise. leaving it entirely in the message won't be helpful.

Reply:
While I'm not sure if I expect developers to actually `rescue` this error, I totally agree that it won't hurt having the record instance passed to the error. Will make a change, thanks!